### DISCUSSION TITLE
How can I use the recently added filter modal?

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#9307] [Help] [answered]
https://github.com/filamentphp/filament/pull/9259

**Comment by sinnbeck** [comment, marked as answer]
Yes. Just update and set `FiltersLayout::Modal`

https://filamentphp.com/docs/3.x/tables/filters#displaying-filters-above-the-table-content

**Reply by ArtMin96** [reply]
Thank you. And how can I make it slide over?

![image](https://github.com/filamentphp/filament/assets/29732308/031be41a-fb30-4252-ac05-9172ea72acce)

**Reply by sinnbeck** [reply]
Just add this before or after the filters
```php
->filtersTriggerAction(fn($action) => $action->slideOver()) //this
->filters([
```
https://filamentphp.com/docs/3.x/tables/filters#customizing-the-filters-dropdown-trigger-action

**Reply by ArtMin96** [reply]
Thank you so much!

**Reply by devhoussam1998** [reply]
when i try to custom the `modalSubmitAction` I get this result:

duplicate reset

<img width="314" alt="Screenshot 2025-01-11 at 15 13 45" src="https://github.com/user-attachments/assets/4aca456a-ea7c-4663-bd41-84fbfd8122de" />

```
->modalSubmitAction(
                        fn (StaticAction $action) => $action
                            ->label(__('Reset'))
                            ->size(ActionSize::Large)
                            ->color('danger')
                            ->icon('heroicon-o-trash')
                            ->extraAttributes(['class' => 'mx-auto'])
                    )
```
